runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -921,20 +925,431 @@ static void VMwrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register
+   auto linkageProperties = linkage->getProperties();
+   intptr_t offsetInDataBlock = isStatic ? offsetof(J9JITWatchedStaticFieldData, fieldAddress): offsetof(J9JITWatchedInstanceFieldData, offset);
+
+   TR::Compilation *comp = cg->comp();
+
+   TR::LabelSymbol* startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol* endLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol* unresolvedLabel = generateLabelSymbol(cg);
+   startLabel->setStartInternalControlFlow();
+   endLabel->setEndInternalControlFlow();
+
+   TR::Register *cpIndexReg = cg->allocateRegister();
+   TR::Register *resultReg = cg->allocateRegister();
+   TR::Register *scratchReg = cg->allocateRegister();
+   
+   // Check if snippet has been loaded
+   if (!isStatic && !static_cast<TR::J9PPCWatchedInstanceFieldSnippet *>(dataSnippet)->isSnippetLoaded() ||
+      (isStatic && !static_cast<TR::J9PPCWatchedStaticFieldSnippet *>(dataSnippet)->isSnippetLoaded()))
+         loadFieldWatchSnippet(cg, node, dataSnippet, dataSnippetRegister, scratchReg, !isStatic);
+
+ 
+   // Setup Dependencies
+   // dataSnippetRegister is always used during OOL sequence.
+   // Requires two argument registers resultReg, and cpIndexReg. 
+   // Static requires an extra arugment fieldClassReg 
+   uint8_t numOfConditions = isStatic? 4 : 3;
+   TR::RegisterDependencyConditions  *deps =  new (cg->trHeapMemory()) TR::RegisterDependencyConditions(numOfConditions, numOfConditions, cg->trMemory());
+   
+   deps->addPreCondition(dataSnippetRegister, TR::RealRegister::NoReg);
+   deps->addPostCondition(dataSnippetRegister, TR::RealRegister::NoReg);
+
+   TR_PPCOutOfLineCodeSection *generateReportOOL = new (cg->trHeapMemory()) TR_PPCOutOfLineCodeSection(unresolvedLabel, endLabel, cg);
+   cg->getPPCOutOfLineCodeSectionList().push_front(generateReportOOL);
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, startLabel);   
+
+   // Compare J9JITWatchedInstanceFieldData.offset or J9JITWatchedStaticFieldData.fieldAddress (Depending on Instance of Static)
+   // Load value from dataSnippetRegister + offsetInDataBlock then compare and branch
+   TR::Register *cndReg = cg->allocateRegister(TR_CCR);
+   generateTrg1MemInstruction(cg, TR::InstOpCode::Op_load, node, scratchReg,
+                     new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetInDataBlock, TR::Compiler->om.sizeofReferenceAddress(), cg));
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, scratchReg, scratchReg, cndReg, -1);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, unresolvedLabel, cndReg);
+   
+   generateReportOOL->swapInstructionListsWithCompilation();
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, unresolvedLabel);
+
+   if (isStatic)
+      {
+      // Fill in J9JITWatchedStaticFieldData.fieldClass
+      TR::Register *fieldClassReg = NULL;
+
+      if (isWrite)
+         {
+         fieldClassReg = cg->allocateRegister();
```

Comment:
There is a problem here. We allocate a register here for `fieldClassReg`. But don't deallocate it using `cg->stopUsingRegister()`. I believe this path of code will also trigger a `TR_ASSERT` in a debug build as this register will still be live.

Reply:
Thanks for catching that! 
I looked around to make sure there weren't any others. 
See
https://github.com/eclipse/openj9/pull/6234/commits/2c18083a93b7d7b0711c9039f7234d65bb9bd742